src/custom-elements/src/components/tabs/tabs.tsx

```diff
@@ -142,7 +146,28 @@ export class Tabs {
   activateTab(tab: TabTrigger, element: HTMLElement) {
     this.activeTab = tab.id;
     this.activeTabElement = element;
-    this.chiTabChange.emit(tab);
+  }
+
+  setActivePanelId(tabs: TabTrigger[]) {
+    tabs.forEach((tab: TabTrigger) => {
+      if (tab.id === this?.activeTab) {
+        this.activePanel = tab.id;
+      } else if (tab.children) {
+        this.setActivePanelId(tab.children);
+      }
+    });
+  }
+
+  activatePanel(panelId?: string): void {
+    const panelItems = this._getPanelItems();
+
+    panelItems.forEach((panel: HTMLElement) => {
+      panel.classList.toggle(ACTIVE_CLASS, panel.id === panelId);
+    });
+  }
+
+  _getPanelItems(): NodeListOf<HTMLElement> {
+    return this.el.querySelectorAll(`.${TABS_CLASSES.PANEL}`);
```

Comment:
Could we avoid iteration by doing:

```suggestion
    // get current active panel, remove active class
    this.el.querySelector(`.${TABS_CLASSES.PANEL} .${ACTIVE_CLASS}`);
    
    // get next active panel, set active class
    this.el.getElementById(panelId).classList.toggle(ACTIVE_CLASS, true);
```

Reply:
Makes sense, done.